pilot/pkg/networking/core/v1alpha3/httproute.go

```diff
@@ -533,8 +533,18 @@ func generateVirtualHostDomains(service *model.Service, port int, node *model.Pr
 // - Given foo.local.campus.net on proxy domain "" or proxy domain example.com, this
 // function returns nil
 func GenerateAltVirtualHosts(hostname string, port int, proxyDomain string) []string {
+	// If the service hostname could possibly be a kube service following the <ns>.svc.<suffix>
+	// naming convention, check the suffix.  Assume that hostnames that match the local dns/proxy domain
+	// are indeed kubernetes services
 	if strings.Contains(proxyDomain, ".svc.") {
-		return generateAltVirtualHostsForKubernetesService(hostname, port, proxyDomain)
+
+		if strings.HasSuffix(hostname, removeSvcNamespace(proxyDomain)) {
+			return generateAltVirtualHostsForKubernetesService(hostname, port, proxyDomain)
+		}
+
+		// Hostname is not a kube service.  Since it contains a `.svc.`, it is not safe to expand the
+		// hostname as it could conflict with other kube services.
```

Comment:
Is this comment correct? How do we know if the host has ".svc"?

Reply:
Actually, it wasn't!  Updated.